pkg/odo/cli/url/create.go

```diff
@@ -66,28 +67,36 @@ func (o *URLCreateOptions) Complete(name string, cmd *cobra.Command, args []stri
 
 // Validate validates the UrlCreateOptions based on completed values
 func (o *URLCreateOptions) Validate() (err error) {
+
 	exists, err := url.Exists(o.Client, o.urlName, o.Component(), o.Application)
 	if exists {
 		return fmt.Errorf("The url %s already exists in the application: %s\n%v", o.urlName, o.Application, err)
 	}
-
 	return
 }
 
 // Run contains the logic for the odo url create command
 func (o *URLCreateOptions) Run() (err error) {
 
-	log.Infof("Adding URL to component: %v", o.Component())
 	urlRoute, err := url.Create(o.Client, o.urlName, o.componentPort, o.Component(), o.Application)
 	if err != nil {
 		return err
 	}
-	urlCreated := url.GetURLString(urlRoute.Spec.Protocol, urlRoute.Spec.URL)
-	log.Successf("URL created for component: %v\n\n"+
-		"%v - %v\n", o.Component(), urlRoute.Name, urlCreated)
+
+	out, err := util.PrintMachineOutput(o.outputFlag, urlRoute)
+	if err != nil {
+		return err
+	}
+	if out != "" {
+		fmt.Println(out)
+	} else {
+
+		log.Successf("URL created for component: %v\n\n"+
+			"%v - %v\n", o.Component(), urlRoute.Name, urlRoute.Spec.URL)
+	}
 
 	if o.urlOpenFlag {
-		err := util.OpenBrowser(urlCreated)
+		err := util.OpenBrowser(urlRoute.Spec.URL)
```

Comment:
This I guess doesn't include the protocol in the URL so I guess it's better to use GetURLString() from above